### ISSUE TITLE
When caps lock is on a tooltip appears in the top left corner of the window on latest macos sonoma

### ISSUE THREAD

**Issue by dortamiguel** [libsdl-org/SDL#8521]
I noticed that macOS added a tooltip to indicate that you focused on a text input with caps lock on. Usually you don't seen it in application without text input, but with SDL is always shown in the top left corner

![image](https://github.com/libsdl-org/SDL/assets/3449006/37c4eebc-468f-485d-b808-f3f807f980c1)

Should be possible somehow to hide it?

**Comment by icculus** [comment]
Hmm, this is probably SDLTranslatorResponder causing this, but I don't see any obviously way to disable this new thing in Cocoa.

```objc
@interface SDLTranslatorResponder : NSView <NSTextInputClient>
```

It can be disabled system-wide with this command line...

```bash
sudo mkdir -p /Library/Preferences/FeatureFlags/Domain
sudo /usr/libexec/PlistBuddy -c "Add 'redesigned_text_cursor:Enabled' bool false" /Library/Preferences/FeatureFlags/Domain/UIKit.plist
```

...but obviously that's not ideal. I'll ask Apple if there's a programmatic way to do it, or if we should just be doing something different.

**Comment by icculus** [comment]
This isn't ideal, but as a quick workaround, if you don't need SDL_TEXTINPUT events, you can call `SDL_StopTextInput` and it will not have the caps lock icon.

(Also, building with an older Xcode flags the binary in a way that tells macOS not to use that icon on your process, but that's not practical overall.)

**Comment by dortamiguel** [comment]
thank you for finding this!

**Comment by icculus** [comment]
I'm leaving this open in case a future release of macOS adds a method to explicitly opt out of the extra UI, but in the short term all we have are workarounds (which, if a better solution arrives, you should discontinue using!).

**Comment by icculus** [comment]
(Bumping this to 3.2.0 milestone since it will be relevant there, too, and so we keep it most visible in the short term. Any fix that arises will be applied to SDL2 as well, presumably.)

**Comment by slouken** [comment]
This is fixed in the latest SDL 3.0 code, thanks!
